### PR TITLE
gobject-introspection: add head, add python3 which requires head 

### PR DESCRIPTION
**Pull request by moyogo**
Python3 support was added on 2015-09-30 in gobject-introspection, after the 1.46.0 release.

### CODE REVIEW

File: Library/Formula/gobject-introspection.rb [hunk 1]
```diff
@@ -17,6 +29,7 @@ class GobjectIntrospection < Formula
   depends_on "cairo"
   depends_on "libffi"
   depends_on "python" if MacOS.version <= :mavericks
+  depends_on "python3" => :optional
```
Comment (DomT4) [inline Library/Formula/gobject-introspection.rb]: Just put this in the `head` block, then you can bin the `if build.with?("python3") && !build.head?` logic and this line. Also `:python3`, not `"python3"`.

**Comment by moyogo** [comment]
@DomT4 Is this better?

**Comment by DomT4** [comment]
You've broken the formula there :wink: -

```
checking whether python1.46.0 version is >= 2.7... no
configure: error: Python interpreter is too old
```

`#{version}` in the context you've used it for stable = formula version, not Python version.

File: Library/Formula/gobject-introspection.rb [hunk 2]
```diff
@@ -27,13 +36,28 @@ def install
...
-    inreplace "configure" do |s|
-      s.change_make_var! "GOBJECT_INTROSPECTION_LIBDIR", "#{HOMEBREW_PREFIX}/lib"
-    end
 
-    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}"
-    system "make"
-    system "make", "install"
+    if build.head?
+      Language::Python.each_python(build) do |python, version|
+        sprintf "HEAD #{python} #{version}"
```
Comment (DomT4) [inline Library/Formula/gobject-introspection.rb]: Not sure why you need to print here.

File: Library/Formula/gobject-introspection.rb [hunk 3]
```diff
@@ -27,13 +36,28 @@ def install
...
+        system "make", "clean"
+      end
+    else
+      inreplace "configure" do |s|
+        s.change_make_var! "GOBJECT_INTROSPECTION_LIBDIR", "#{HOMEBREW_PREFIX}/lib"
+      end
+      system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}", "PYTHON=python#{version}"
+      system "make"
+      system "make", "install"
+      system "make", "clean"
```
Comment (DomT4) [inline Library/Formula/gobject-introspection.rb]: You shouldn't need to `make clean`. The build directory is disposed of after use.

File: Library/Formula/gobject-introspection.rb [hunk 4]
```diff
@@ -27,13 +36,28 @@ def install
...
+      Language::Python.each_python(build) do |python, version|
+        sprintf "HEAD #{python} #{version}"
+        system "./autogen.sh"
+        inreplace "configure" do |s|
+          s.change_make_var! "GOBJECT_INTROSPECTION_LIBDIR", "#{HOMEBREW_PREFIX}/lib"
+        end
+        system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}", "PYTHON=python#{version}"
+        system "make"
+        system "make", "install"
+        system "make", "clean"
```
Comment (DomT4) [inline Library/Formula/gobject-introspection.rb]: You shouldn't need to `make clean`. The build directory is disposed of after use.

**Comment by bfontaine** [comment]
This PR needs to be rebased on top of `master`.

**Comment by bfontaine** [comment]
Closing due to no response. Feel free to open another PR addressing the comments above. Thanks!

**Pull request closed without merging**